### PR TITLE
Feature create edit indicator #133

### PR DESCRIPTION
**Pull request by lkolacz**
##### Feature list
* createan indicator in cluster reporting

* update indicator in cluster reporting

##### Progress checker
- [x] Django

- [x] Polymer

- [x] Django test cases

- [ ] ~Polymer test cases~

### CODE REVIEW

**Comment by n1207n** [comment]
@chachra,

@lkolacz and I had a standup discussion today where we are not sure how updating Indicator would work.

The questions are:
* For updating Indicator, can locations be updated?
* If locations for Indicator can be updated, then would there be a case where 2 different Indicators having same location? If this is true then we need to make locations to ManyToManyField. Or once the location is taken by one Indicator, is it illegal for another Indicator to link the taken location? Then we can keep our ForeignKey in Location.

**Comment by n1207n** [comment]
@chachra whenever you are free. Take a look on above questions.

**Comment by chachra** [comment]
@n1207n @lkolacz

- yes locations can be updated
- yes locations can be shared... since there are multiple locations per indicator I assumed from indicator (reportable) -> locations we have a M2M... 
unicef_humanitarian_cluster_portal_kickoff_-_etools_prp__lucidchart
- frequency lets go with weekly, weekly, monthly for now. There is a "specific dates" option which might require a m2m or json or list field. I can't find mockup for that anymore @kasialanecka - please post here.

**Comment by chachra** [comment]
@n1207n @lkolacz @kasialanecka 
pmp-pm-changes

in PMP its done at PD level... but its needed in the PRP model (and hence the indicator add/edit modal as well). This is 4th option for reporting. Can do in this ticket or open separate one. I think using a list field for the dates might work...

**Comment by kasialanecka** [comment]
@lkolacz @n1207n @chachra 
there is a comment in invision about this field:

screen shot 2017-06-26 at 14 29 09

**Comment by chachra** [comment]
@lkolacz - please move all changes / fixes relating to edit to this branch. And please close out the other PR.

**Comment by lkolacz** [comment]
@chachra DONE

File: django_api/django_api/apps/indicator/tests/test_views.py [hunk 1]
```diff
@@ -235,6 +231,315 @@ def test_list_api_with_limit(self):
             self.assertEquals(len(response.data), 2)
 
 
+# TODO: Re-enable this test case at #35 branch
```
Comment (n1207n) [inline django_api/django_api/apps/indicator/tests/test_views.py]: This is old code ;) Remove this commented section.

**Comment by n1207n** [comment]
Ignore above comment. I just understood that location needs to be nullable so that reportable can be updated to different location.

File: django_api/django_api/apps/core/migrations/0004_merge_20170708_1618.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
```
Comment (lkolacz) [inline django_api/django_api/apps/core/migrations/0004_merge_20170708_1618.py]: done

**Comment by rkrupinski** [comment]
@ErinOC Check https://github.com/unicef/etools-partner-reporting-portal/issues/133 for details on how to use the `<indicator-modal>` component. Also, check with @lkolacz on where to get `object_id` & `object_type` from.

File: polymer/bower.json [hunk 3]
```diff
@@ -66,6 +67,10 @@
     "paper-checkbox": "~1.0.13",
     "paper-menu-button": "^1.3.0",
     "moment-element": "^1.1.1",
-    "paper-dialog": "^1.1.0"
+    "paper-dialog": "^1.1.0",
```
Comment (ErinOC) [inline polymer/bower.json]: I had to change this to "paper-dialog": "^2.0.0" before the Docker container would build.

File: polymer/src/elements/cluster-reporting/disaggregations-widget.html [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+
+      observers: [
+        '_setValidity(value.splices)',
+        '_setCanAddMore(value.splices)',
+      ],
+
+      _setValidity: function () {
+        var invalid;
+
+        if (!this.value.length) {
```
Comment (ErinOC) [inline polymer/src/elements/cluster-reporting/disaggregations-widget.html]: Don't we want to allow the option to have zero disaggregations?

Comment (rkrupinski) [inline polymer/src/elements/cluster-reporting/disaggregations-widget.html]: That's a good question, actually.

@chachra ?

File: polymer/bower.json (hunk 3 above)
Comment (rkrupinski) [inline polymer/bower.json]: @ErinOC Please remove your (polymer) docker image and try again. `2.0` has breaking changes (as the version number implies) and we'd rather avoid switching.

Comment (ErinOC) [inline polymer/bower.json]: @rkrupinski I deleted my image and tried again, and got the same error.  I asked Silin to try it too just to make sure it wasn't something weird on my end, and he had the same result:
image

Comment (rkrupinski) [inline polymer/bower.json]: Bumped `paper-dialog` to `^2.0.0`.

**Pull request merged**